### PR TITLE
Feature/extending dashboard with apps

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it introduces the possibility to extend the dashboard with apps.

# Impact

* [x] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -131,6 +131,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Add app support for events - #7622 by @IKarbowiak
 - Add date & date time attributes - #7500 by @piotrgrundas
 - Add `withChoices` flag for Attribute type - #7733 by @dexon44
+- Extend app by `AppExtension` - #7701 by @korycins
```
Comment (maarcingebala) [inline CHANGELOG.md:134]: This entry should be moved to the 3.1 section.

**Review by IKarbowiak — APPROVED** [review]

File: saleor/app/types.py [hunk 2]
```diff
@@ -3,3 +3,40 @@ class AppType:
...
+    PRODUCT - app's iframe will be mounted in product section
+    """
+
+    PRODUCT = "product"
+
+    CHOICES = [(PRODUCT, "product")]
+
+
+class AppExtensionType:
+    """All available type where app's iframe can be mounted.
```
Comment (maarcingebala) [inline saleor/app/types.py:20]:
```suggestion
    """All available types where app's iframe can be mounted.
```

File: saleor/app/types.py [hunk 3]
```diff
@@ -3,3 +3,40 @@ class AppType:
     THIRDPARTY = "thirdparty"
 
     CHOICES = [(LOCAL, "local"), (THIRDPARTY, "thirdparty")]
+
+
+class AppExtensionView:
+    """All available place where app's iframe can be mounted.
```
Comment (maarcingebala) [inline saleor/app/types.py:9]:
```suggestion
    """All available places where app's iframe can be mounted.
```

File: saleor/graphql/app/types.py [hunk 4]
```diff
@@ -2,15 +2,87 @@
...
+
+
+class AppManifestExtension(graphene.ObjectType):
+    permissions = graphene.List(
+        graphene.NonNull(Permission),
+        description="List of the app extension's permissions.",
+        required=True,
+    )
+    label = graphene.String(
+        description="Label of the extension show on dashboard side.", required=True
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:33]:
```suggestion
        description="Label of the extension to show in the dashboard.", required=True
```

File: saleor/graphql/app/types.py [hunk 5]
```diff
@@ -2,15 +2,87 @@
...
+    )
+    url = graphene.String(
+        description="URL where iframe of extension is placed.", required=True
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:36]: Is it the URL of a view?
```suggestion
        description="URL of a view where extension's iframe is placed.", required=True
```

File: saleor/graphql/app/types.py [hunk 6]
```diff
@@ -2,15 +2,87 @@
...
+    )
+    view = AppExtensionViewEnum(
+        description="View where extension's iframe will be mounted.", required=True
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:39]:
```suggestion
        description="Name of a view where extension's iframe will be mounted.", required=True
```

File: saleor/graphql/app/types.py [hunk 7]
```diff
@@ -2,15 +2,87 @@
...
+    )
+    type = AppExtensionTypeEnum(
+        description="Type of view where extension's iframe will be mounted.",
```
Comment (maarcingebala) [inline saleor/graphql/app/types.py:42]:
```suggestion
        description="Type of a view where extension's iframe will be mounted.",
```

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**